src/main/java/duke/Task.java

```diff
@@ -0,0 +1,65 @@
+package duke;
+
+/**
+ * Task class to represent a task, which is a todo, deadline or event.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected char type;
+
+    /**
+     * Constructor for Task class.
+     * Initializes a Task object with given description and marks it as not done.
+     * @param description task description
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns an icon depending on whether this task is done.
+     * @return X if done or blank if not done.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); //return X or blank space, X means done
```

Comment:
im not sure, but the eg given in Java coding standards seems to do something like (isDone) ? "X" : " ";